chain/consensus/mir/crypto.go

```diff
@@ -0,0 +1,124 @@
+package mir
+
+import (
+	"context"
+	"crypto/sha256"
+	"strings"
+
+	"golang.org/x/xerrors"
+
+	"github.com/filecoin-project/go-address"
+	filcrypto "github.com/filecoin-project/go-state-types/crypto"
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/lib/sigs"
+	mir "github.com/filecoin-project/mir/pkg/modules"
+	t "github.com/filecoin-project/mir/pkg/types"
+)
+
+var MsgMeta = api.MsgMeta{Type: "mir-request"}
+
+var _ mir.Crypto = &CryptoManager{}
+
+type CryptoManager struct {
+	addr   address.Address
+	wallet api.Wallet
+}
+
+func NewCryptoManager(addr address.Address, wallet api.Wallet) (*CryptoManager, error) {
+	if addr.Protocol() != address.SECP256K1 {
+		return nil, xerrors.New("must be SECP address")
+	}
+	return &CryptoManager{addr, wallet}, nil
+}
+
+// Sign signs the provided data and returns the resulting signature.
+// The data to be signed is the concatenation of all the passed byte slices.
+// A signature produced by Sign is verifiable using VerifyNodeSig or VerifyClientSig,
+// if, respectively, RegisterNodeKey or RegisterClientKey has been invoked with the corresponding public key.
+// Note that the private key used to produce the signature cannot be set ("registered") through this interface.
+// Storing and using the private key is completely implementation-dependent.
+func (c *CryptoManager) Sign(data [][]byte) (bytes []byte, err error) {
+	signature, err := c.wallet.WalletSign(context.Background(), c.addr, hash(data), MsgMeta)
```

Comment:
Why use a plain hash instead of a `multihash` or a `cid`?

Reply:
Whatever we sign, it has to be computed locally, and I find the solution with the plain hash good for now, mostly because it is simple. Later, when we want to support multiple different implementation s of the crypto module, we'll need to specify exactly how the signature over the given data is computed, including what kind of hash to use (if any). As long all crypto modules return the same signature over the same data, we're good. And I'd say that for now it's ok to count on a single simple implementation.